lib/IRGen/GenHeap.cpp

```diff
@@ -1244,6 +1244,32 @@ llvm::Constant *IRGenModule::getFixLifetimeFn() {
   return fixLifetime;
 }
 
+llvm::Constant *IRGenModule::getFixedClassInitializationFn() {
+  if (FixedClassInitializationFn)
+    return *FixedClassInitializationFn;
+  
+  // If ObjC interop is disabled, we don't need to do fixed class
+  // initialization.
+  llvm::Constant *fn;
+  if (!ObjCInterop) {
```

Comment:
I think that this could be hoisted further:

```
  if (FixedClassInitializationFn)
    return *FixedClassInitializationFn;

  if (!ObjcInterop)
    return nullptr;

  if (getAvailabilityContext().isContainedIn(Context.getSwift51Availability()))
    FixedClassInitializationFn = getObjCOptSelFn();
  else
    FixedClassInitializationFn = getGetInitializedObjCClassFn();

  return FixedClassInitializationFn;
```

Reply:
Because FixedClassInitializationFn is  Optional, and is a nonlocal instance variable, I think it’s preferable to keep the new value in a nonoptional local variable and assign to global state with a weaker type once.